src/drivers/gitlab.js

```diff
@@ -19,39 +19,45 @@ class Gitlab {
 
     this.token = token;
     this.repo = repo;
+  }
+
+  async project_path() {
+    const api_path = await this.detect_api_v4();
+    const project_path = encodeURIComponent(
+      this.repo.replace(api_path, '').substr(1)
+    );
 
-    const { protocol, host, pathname } = new URL(this.repo);
-    this.project_path = encodeURIComponent(pathname.substring(1));
-    this.repo_origin = `${protocol}//${host}`;
+    return project_path;
   }
 
-  async detect_api_v4(opts = {}) {
-    const { pathname } = new URL(this.repo);
+  async detect_api_v4() {
```

Comment:
After your modifications, `detect_api_v4` should probably be renamed to `detect_base` or something similar.

Reply:
I thought the same but it uses '/api/v4/version'
Im going to setup this hold on

Reply:
Note: I chose `detect_base` instead of `detect_api` because we're actually returning the GitLab server base, and not the API base.

Reply:
In the same line, `repo_base` would also be a bit of a stretch, as the address we return represents the GitLab server base. What about `server_base` instead? By the way: this is just nitpicking, please ignore it if you want to.

Reply:
I prefer repo_base to be aligned with repo

Reply:
👌

Reply:
Please keep in mind that this address doesn't necessarily involve a `repo` context and, sometimes, we're only being fed with an orgainzation address.

Reply:
I get you, but its an internal variable and sounded to me nicer to relate it to the source.
If weeks later I read repo_base I will relate it to repo to some extend while server_base I will to figure out what was that

Reply:
👌